### ISSUE TITLE
Framework-level Identity Management

### ISSUE THREAD

**Issue by onethirtyfive** [padrino/padrino-framework#133]
I was speaking with Nathan about implementing a flexible, role-based authorization system, and the conversation branched into authentication as well.

I mentioned that, on my site, registration happens in two steps: first, the user verifies his or her identity via any number of schemes (Email, OpenID, etc...), and then goes on to register as a user. It occurred to me that--in theory at least--authentication is supposed to happen on a protocol level. Barring that, it would still be beneficial to handle it via middleware. It should be separate from authorization to perform an action on a resource.

In other words, some routes need authentication to establish context. Period. For these routes, I propose we provide a generator that has robust Identity models for different authorization schemes. They should also provide configurable authorization logic (like a redirect URL for OpenID, for example).

These identities can be linked with the site's user account as per settings, and then the developer would have the option of implementing only account-specific stuff, and leaving the identity management to padrino. At least to get going.

The web is certainly moving in this direction--everyone has multiple account fatigue.  Previous attempts such as Rails' restful_authentication failed because they conflate the concerns of authentication with authorization. These two separate concerns can't be done generally enough to accommodate all domains.

Authorization, however, almost certainly can be done generally enough.

tl;dr: let's have padrino (optionally) manage user identities.
